### ISSUE TITLE
Printing listmode coincidences for GE Signa prints wrong data

### ISSUE THREAD

**Issue by AnderBiguri** [UCL/STIR#573]
When doing 

`list_lm_events --coincidence 1 ./LIST0000.BLF`

for GE Signa RDF9 type of ListMode data, prints stuff like: 

```
Coincidence p (c:4294967083,r:57,l:0)-(c:4294966846,r:42,l:0)
Coincidence p (c:171,r:24,l:0)-(c:4294966862,r:12,l:0)
Coincidence p (c:427,r:8,l:0)-(c:390,r:5,l:0)
Coincidence p (c:4294967211,r:48,l:0)-(c:298,r:37,l:0)
Time 2164260899
Coincidence p (c:4294966902,r:61,l:0)-(c:447,r:0,l:0)
Coincidence p (c:4294966916,r:62,l:0)-(c:379,r:15,l:0)
Coincidence p (c:4294966724,r:62,l:0)-(c:344,r:31,l:0)
Coincidence p (c:4294966784,r:36,l:0)-(c:4294966724,r:62,l:0)
Coincidence p (c:376,r:31,l:0)-(c:4294966784,r:47,l:0)
Coincidence p (c:168,r:31,l:0)-(c:4294966840,r:62,l:0)
```

Clearly some of those numbers are not OK (we wish we had a detector with 4294966916 crystals! the resolution!). 

I am unsure if the issue is the printing, the storage or the reading of RDF9 data. 

Pinging @pw351 and @KrisThielemans to let you know I opened an issue.

**Comment by pwadhwa351** [comment]
I think the LIST0000.BLF file that you are using is compressed. Can you confirm if it is compressed or uncompressed? 
Also I had a quick look with one of my data, I get this:
Coincidence p (c:437,r:41,l:0)-(c:177,r:16,l:0) delta time: -338.52
Coincidence p (c:48,r:19,l:0)-(c:339,r:11,l:0) delta time: 2083.2
Coincidence p (c:338,r:31,l:0)-(c:191,r:18,l:0) delta time: -911.4
Coincidence p (c:388,r:38,l:0)-(c:122,r:27,l:0) delta time: -143.22
Coincidence p (c:134,r:15,l:0)-(c:379,r:22,l:0) delta time: 768.18
Coincidence p (c:152,r:13,l:0)-(c:398,r:27,l:0) delta time: 755.16
Coincidence p (c:151,r:31,l:0)-(c:358,r:38,l:0) delta time: 1822.8
Coincidence p (c:115,r:9,l:0)-(c:420,r:9,l:0) delta time: 78.12
Coincidence p (c:271,r:15,l:0)-(c:22,r:13,l:0) delta time: -494.76
Coincidence p (c:37,r:28,l:0)-(c:247,r:17,l:0) delta time: 1002.54
Coincidence p (c:185,r:35,l:0)-(c:15,r:17,l:0) delta time: -1067.64
Coincidence p (c:49,r:2,l:0)-(c:240,r:12,l:0) delta time: 403.62
Coincidence p (c:113,r:4,l:0)-(c:253,r:38,l:0) delta time: 455.7
Coincidence p (c:225,r:10,l:0)-(c:29,r:36,l:0) delta time: -182.28
Coincidence p (c:206,r:32,l:0)-(c:444,r:38,l:0) delta time: 195.3
Coincidence p (c:53,r:10,l:0)-(c:221,r:19,l:0) delta time: 390.6
Coincidence p (c:427,r:3,l:0)-(c:168,r:14,l:0) delta time: -78.12
Coincidence p (c:18,r:29,l:0)-(c:228,r:15,l:0) delta time: 65.1
Coincidence p (c:276,r:13,l:0)-(c:214,r:26,l:0) delta time: -637.98
Coincidence p (c:59,r:21,l:0)-(c:184,r:37,l:0) delta time: 2122.26
Coincidence p (c:21,r:17,l:0)-(c:165,r:21,l:0) delta time: 690.06
Coincidence p (c:126,r:19,l:0)-(c:362,r:9,l:0) delta time: 559.86
Coincidence p (c:358,r:39,l:0)-(c:254,r:14,l:0) delta time: -78.12
Coincidence p (c:284,r:21,l:0)-(c:29,r:11,l:0) delta time: -846.3
Coincidence p (c:11,r:12,l:0)-(c:143,r:34,l:0) delta time: 1900.92
Coincidence p (c:68,r:20,l:0)-(c:395,r:37,l:0) delta time: 677.04
Coincidence p (c:342,r:37,l:0)-(c:248,r:9,l:0) delta time: -351.54
Coincidence p (c:309,r:2,l:0)-(c:89,r:19,l:0) delta time: -364.56
Coincidence p (c:248,r:10,l:0)-(c:444,r:24,l:0) delta time: 820.26
Coincidence p (c:305,r:23,l:0)-(c:34,r:15,l:0) delta time: -637.98
Coincidence p (c:196,r:39,l:0)-(c:33,r:5,l:0) delta time: -65.1
Coincidence p (c:21,r:29,l:0)-(c:270,r:28,l:0) delta time: 325.5
Coincidence p (c:209,r:25,l:0)-(c:432,r:42,l:0) delta time: 325.5
Coincidence p (c:27,r:10,l:0)-(c:235,r:6,l:0) delta time: 273.42
Coincidence p (c:89,r:10,l:0)-(c:248,r:13,l:0) delta time: 1080.66
Coincidence p (c:292,r:18,l:0)-(c:436,r:6,l:0) delta time: 156.24
Time 4288
Coincidence p (c:357,r:11,l:0)-(c:181,r:24,l:0) delta time: -91.14
Coincidence p (c:88,r:13,l:0)-(c:277,r:21,l:0) delta time: 1093.68
Coincidence p (c:447,r:3,l:0)-(c:203,r:14,l:0) delta time: -377.58
Coincidence p (c:440,r:36,l:0)-(c:216,r:21,l:0) delta time: -299.46
Coincidence p (c:241,r:22,l:0)-(c:17,r:30,l:0) delta time: -26.04
Coincidence p (c:272,r:7,l:0)-(c:323,r:7,l:0) delta time: 1197.84
Coincidence p (c:119,r:1,l:0)-(c:323,r:7,l:0) delta time: 260.4
Coincidence p (c:272,r:7,l:0)-(c:119,r:1,l:0) delta time: -937.44
Coincidence p (c:378,r:40,l:0)-(c:191,r:6,l:0) delta time: -651
Coincidence p (c:84,r:40,l:0)-(c:263,r:10,l:0) delta time: 442.68
Coincidence p (c:150,r:6,l:0)-(c:423,r:31,l:0) delta time: 481.74
Coincidence p (c:17,r:23,l:0)-(c:221,r:37,l:0) delta time: 364.56
Coincidence p (c:401,r:37,l:0)-(c:183,r:7,l:0) delta time: -729.12
Coincidence p (c:317,r:34,l:0)-(c:178,r:43,l:0) delta time: -1080.66
Coincidence p (c:141,r:3,l:0)-(c:335,r:37,l:0) delta time: 1835.82
Coincidence p (c:202,r:12,l:0)-(c:22,r:3,l:0) delta time: -195.3
Coincidence p (c:392,r:1,l:0)-(c:251,r:22,l:0) delta time: -585.9
Coincidence p (c:245,r:15,l:0)-(c:75,r:25,l:0) delta time: -13.02
Coincidence p (c:149,r:34,l:0)-(c:407,r:11,l:0) delta time: 559.86
Coincidence p (c:136,r:36,l:0)-(c:380,r:34,l:0) delta time: 859.32
Coincidence p (c:210,r:33,l:0)-(c:66,r:38,l:0) delta time: -260.4
Coincidence p (c:150,r:18,l:0)-(c:400,r:38,l:0) delta time: 898.38
Coincidence p (c:53,r:7,l:0)-(c:124,r:33,l:0) delta time: 182.28
Coincidence p (c:58,r:1,l:0)-(c:330,r:5,l:0) delta time: 65.1
Coincidence p (c:238,r:23,l:0)-(c:60,r:39,l:0) delta time: -364.56
Coincidence p (c:328,r:8,l:0)-(c:102,r:42,l:0) delta time: -559.86
Coincidence p (c:44,r:19,l:0)-(c:102,r:42,l:0) delta time: 755.16
Coincidence p (c:44,r:19,l:0)-(c:328,r:8,l:0) delta time: 195.3
Coincidence p (c:128,r:38,l:0)-(c:392,r:8,l:0) delta time: 703.08
Coincidence p (c:377,r:31,l:0)-(c:195,r:43,l:0) delta time: -377.58
Coincidence p (c:235,r:3,l:0)-(c:83,r:5,l:0) delta time: -104.16
Coincidence p (c:32,r:5,l:0)-(c:218,r:21,l:0) delta time: 299.46
Coincidence p (c:87,r:32,l:0)-(c:370,r:12,l:0) delta time: 1692.6
Coincidence p (c:134,r:32,l:0)-(c:385,r:1,l:0) delta time: 533.82
Coincidence p (c:246,r:3,l:0)-(c:31,r:35,l:0) delta time: -156.24
Coincidence p (c:180,r:9,l:0)-(c:434,r:14,l:0) delta time: 39.06
Coincidence p (c:340,r:16,l:0)-(c:214,r:20,l:0) delta time: -703.08
Coincidence p (c:58,r:20,l:0)-(c:268,r:7,l:0) delta time: 52.08
Coincidence p (c:234,r:14,l:0)-(c:420,r:12,l:0) delta time: 143.22
Coincidence p (c:82,r:3,l:0)-(c:202,r:41,l:0) delta time: 130.2
Coincidence p (c:48,r:19,l:0)-(c:268,r:8,l:0) delta time: 703.08
Coincidence p (c:308,r:35,l:0)-(c:146,r:44,l:0) delta time: -1236.9
Coincidence p (c:128,r:14,l:0)-(c:377,r:3,l:0) delta time: 455.7
Coincidence p (c:129,r:43,l:0)-(c:411,r:33,l:0) delta time: 833.28
Coincidence p (c:93,r:1,l:0)-(c:273,r:28,l:0) delta time: 559.86
Coincidence p (c:233,r:31,l:0)-(c:424,r:39,l:0) delta time: 716.1
Coincidence p (c:107,r:36,l:0)-(c:223,r:11,l:0) delta time: 1249.92
Coincidence p (c:377,r:33,l:0)-(c:223,r:11,l:0) delta time: -351.54
Coincidence p (c:107,r:36,l:0)-(c:377,r:33,l:0) delta time: 898.38
Coincidence p (c:419,r:13,l:0)-(c:352,r:12,l:0) delta time: -1015.56
Coincidence p (c:352,r:12,l:0)-(c:412,r:20,l:0) delta time: 1080.66

I hope this helps!

**Comment by AnderBiguri** [comment]
@pwadhwa351 ah I see. `h5read(BLF, '/HeaderData/ListHeader/isListCompressed')` returns true in MATLAB. I didn't even know this was an option... How do I decompress it?

**Comment by AnderBiguri** [comment]
This has been solved because my data was compressed, but I would leave it open as a  "we need to check if the file is compressed"

**Comment by KrisThielemans** [comment]
merged into #601
